engine/runtime/src/main/java/org/enso/interpreter/runtime/state/IOPermissions.java

```diff
@@ -0,0 +1,31 @@
+package org.enso.interpreter.runtime.state;
+
+public record IOPermissions(String name, boolean isInputAllowed, boolean isOutputAllowed) {
```

Comment:
Using `record` as `enum`! You don't want this class to have public constructor. Try:
```
private IOPermissions {
}
```